app/models/api_keys.rb

```diff
@@ -16,14 +16,15 @@ def extract_keys_from_string(keys_string)
     keys_string.split(';').map do |key_params|
       begin
         key = JSON.parse(key_params)
-        ApiKey.new(key['key'], key['app_name'], key['env']) if validate_api_key_hash(key)
+        ApiKey.new(key['key'], key['private_key'], key['app_name'], key['env']) if validate_api_key_hash(key)
       rescue
         next
       end
     end.compact
   end
 
   def find_api_key(key)
+    return unless @api_keys
```

Comment:
if no api-keys are valid (initially), there would be an error, because `@api_keys.find` would produce a nil error, because nil.find doesn't exist. do you think we can get rid of it?